operator/check_operator.go

```diff
@@ -52,9 +52,9 @@ func (c operatorCheck) Run(ctx context.Context) (runtime.Results, error) {
 		lib.LogThroughArtifactWriterIfSet(ctx)
 	}
 
-	// NOTE(from Jose): Workaround for DeployableByOLM which relies on ctrl.GetConfig()
-	if _, isSet := os.LookupEnv("KUBECONFIG"); !isSet {
-		os.Setenv("KUBECONFIG", c.kubeconfig)
+	kubeconfig, err := io.ReadAll(c.kubeconfig)
```

Comment:
Is there any reason to delay this read to here? What do we gain by passing the io.Reader instead of just the byte slice? I think if it's ultimately just going to be the byte slice, passing the io.Reader seems an unnecessary abstraction. It doesn't change the testability.

Reply:
Fair. I've changed it to accept just the byte slice instead.